lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -523,6 +524,48 @@ class ConvertAtenMaxPool2dWithIndicesBackwardOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Value input = adaptor.self();
+    auto resultType = input.getType().cast<RankedTensorType>();
+    Type elementType = resultType.getElementType();
+    Location loc = op->getLoc();
+    SmallVector<Value> sizes = getTensorSizes(rewriter, loc, input);
+    Value output = createZeroInitTensor(rewriter, loc, sizes, elementType);
+    output = rewriter.create<tensor::CastOp>(loc, resultType, output);
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant dim value is supported");
+
+    sizes.erase(sizes.begin() + dim);
+    SmallVector<int64_t> accStatic(resultType.getShape());
+    accStatic.erase(accStatic.begin() + dim);
+    Value acc = createZeroInitTensor(rewriter, loc, sizes, elementType);
+    Type accType = RankedTensorType::get(accStatic, elementType);
+    acc = rewriter.create<tensor::CastOp>(loc, accType, acc);
+
+    Value result =
+        rewriter
+            .create<TMTensor::ScanOp>(loc, TypeRange{resultType, acc.getType()},
```

Comment:
For the `TMTensor` ops, you need to create a block inside the region of the op with the computation that needs to be performed as the scan takes place. For example, this is how a `TMTensor::ScatterOp` is created:

https://github.com/llvm/torch-mlir/blob/4d693df63d9dd325a69d22f9180695349cd6771c/lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp#L64-L65

The test is currently segfaulting because when the code that lowers the `ScanOp` tries to access the block inside the region of the op, there is no block to access.

You can look at the lit tests of `TMTensor` to see an [example of what the `ScanOp` should look like](https://github.com/llvm/torch-mlir/blob/e52e8868456ce3c99091f501c1c520989785342c/externals/llvm-external-projects/torch-mlir-dialects/test/tmtensor/convert_to_loops.mlir#L69).

Reply:
Ah, I see. I've added another helper like that for Scan, that seems to have done the trick.